url64d634b8f0_agdturner_generic_tgz-pJ8-uk_ac_leeds_ccg_andyt_generic_visualisation_charts_Generic_BarChartJ8/src/uk/ac/leeds/ccg/andyt/generic/utilities/Generic_Time.java

```diff
@@ -0,0 +1,234 @@
+            int year) {
+        this.day = day;
+        this.month = month;
+        this.year = year;
+    }
+    
+    public static int getAgeInYears(
+            Generic_Time timeNow,
+            Generic_Time timeCreated) {
+        if (timeNow.month > timeCreated.month) {
+            return timeNow.year - timeCreated.year;
+        } else {
+            if (timeNow.month == timeCreated.month) {
+                if (timeNow.day >= timeCreated.day) {
+                    return timeNow.year - timeCreated.year;
+                } else {
+                    return timeNow.year - timeCreated.year - 1;
+                }
+            } else {
+                return timeNow.year - timeCreated.year - 1;
+            }
+        }
+    }
+    
+
+    public static void printTime(long millis) {
+        int MilliSecondsInDay = 24 * MilliSecondsInHour;
+        int MilliSecondsInMinute = 60000;
+        long millis2 = millis;
+        int days = 0;
+        while (millis2 > MilliSecondsInDay) {
+            days++;
+            millis2 -= MilliSecondsInDay;
+        }
+        int hours = 0;
+        while (millis2 > MilliSecondsInHour) {
+            hours++;
+            millis2 -= MilliSecondsInHour;
+        }
+        int minutes = 0;
+        while (millis2 > MilliSecondsInMinute) {
+            minutes++;
+            millis2 -= MilliSecondsInMinute;
+        }
+        int seconds = 0;
+        while (millis2 > 1000) {
+            seconds++;
+            millis2 -= MilliSecondsInSecond;
+        }
+        System.out.println("" + days + " day(s) "
+                + hours + " hour(s) "
+                + minutes + " minute(s) "
+                + seconds + " second(s) "
+                + millis2 + " millisecond(s)");
+    }
+
+    /**
+     * @return      * {@code
+    ArrayList<String> result;
+     * result = new ArrayList<String>();
+     * result.add("Jan");
+     * result.add("Feb");
+     * result.add("Mar");
+     * result.add("Apr");
+     * result.add("May");
+     * result.add("Jun");
+     * result.add("Jul");
+     * result.add("Aug");
+     * result.add("Sep");
+     * result.add("Oct");
+     * result.add("Nov");
+     * result.add("Dec");
+     * return result;
+     * }
+     */
+    public static ArrayList<String> getMonths3Letters() {
+        ArrayList<String> result;
+        result = new ArrayList<String>();
+        result.add("Jan");
+        result.add("Feb");
+        result.add("Mar");
+        result.add("Apr");
+        result.add("May");
+        result.add("Jun");
+        result.add("Jul");
+        result.add("Aug");
+        result.add("Sep");
+        result.add("Oct");
+        result.add("Nov");
+        result.add("Dec");
+        return result;
+    }
+
+    public static int getMonth(
+            String month,
+            ArrayList<String> month3Letters) {
+        return month3Letters.indexOf(month.substring(0, 3)) + 1;
+    }
+
+    public static String getMonth3Letters(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.